rust/gitxetcore/src/command/materialize.rs

```diff
@@ -0,0 +1,60 @@
+use clap::Args;
+use lazy::lazy_pathlist_config::{check_or_create_lazy_config, LazyPathListConfigFile};
+
+use crate::errors::{GitXetRepoError, Result};
+use crate::git_integration::git_repo::GitRepo;
+use crate::git_integration::git_wrap::{self, list_files_from_repo};
+use crate::{config::XetConfig, constants::GIT_LAZY_CHECKOUT_CONFIG};
+
+#[derive(Args, Debug)]
+pub struct MaterializeArgs {
+    #[clap(short, long)]
+    recursive: bool,
+
+    path: String,
+}
+
+pub async fn materialize_command(cfg: XetConfig, args: &MaterializeArgs) -> Result<()> {
+    // Make sure repo working directory is clean
+    let repo = GitRepo::open(cfg.clone())?;
+
+    eprintln!("Checking repo state...");
```

Comment:
This command should be nearly instantaneous, so I think we can drop the print statement here (we don't print anything elsewhere when we check).

Reply:
Sounds good will remove.